src/hal/arm/exceptions.c

```diff
@@ -78,6 +81,16 @@ static int exceptions_i2s(char *prefix, char *s, unsigned int i, unsigned char b
 }
 
 
+static char *exceptions_cpy(char *s1, const char *s2)
+{
+	while ((*s1++ = *s2++))
+		;
+	s1--;
+
+	return s1;
```

Comment:
Maybe 
`return s1 - 1;` 
for compactness?

Reply:
Anyway, should we really subtract one?
buff = exceptions_cpy(buff, "\nException: ");

*buff now contains _space_ (char after : and before '\0') in this version, which we will overwrite in next invocation of exceptions_cpy.

Reply:
`exceptions_cpy` is implementation of `stpcpy`

after the loop 
` while ((*s1++ = *s2++))`
the pointer s1 is set to the character after '\0', so we need to subtract one to point to the character '\0'

`return s1-1`
this version is probably bad because return value and s1 are different.
In my code I my use this functon as
`exceptiions_cpy(s1, s2)` and next use s1
or
`buf = exceptions(s1, s2)` and use buf

What do you think about adding the `hal_stpcpy` function to string.h.

Reply:
> exceptions_cpy is implementation of stpcpy

after the loop
while ((*s1++ = *s2++))
the pointer s1 is set to the character after '\0', so we need to subtract one to point to the character '\0'

Yes, you're right, my mistake.

I think that

```
s1--;
return s1;
```

is equivalent to
`return s1 - 1;`
or
`return --s1;`

or am I missing something? We don't really care about s1 value on the end, as it is an local variable.

Reply:
Yes, you're right, my mistake, s1 is local variable
Yes, these three versions are the equivalent.